src/common/Web3Client.ts

```diff
@@ -91,6 +99,20 @@ export default class Web3Client {
     return this.web3
   }
 
+  get networkAgnosticWeb3() {
+    if (!this.networkAgnostic) {
+      throw new Error(`API KEY is not define`)
+    }
+    this.biconomy
+      .onEvent(this.biconomy.READY, () => {
+        console.log('Mexa is Ready') // eslint-disable-line
+      })
+      .onEvent(this.biconomy.ERROR, (error, message) => {
+        console.error(error, message) // eslint-disable-line
+      })
```

Comment:
The getter is executed each time it is accessed. So it is not a good idea to subscribe to events here. The subscriptions will add up with time and affect performance.

Reply:
Also, no need to log things in happy state.

Reply:
updated to check while defining in constructor only.